src/Bicep.Core/Syntax/ModuleDeclarationSyntax.cs

```diff
@@ -40,5 +42,29 @@ public ModuleDeclarationSyntax(Token keyword, IdentifierSyntax name, SyntaxBase
         public override TextSpan Span => TextSpan.Between(Keyword, Body);
 
         public StringSyntax? TryGetPath() => Path as StringSyntax;
+
+        public TypeSymbol GetDeclaredType(SemanticModel.SemanticModel moduleSemanticModel)
```

Comment:
I didn't see a good place to put these, so I put them on the syntax nodes. Open to better ideas. The methods to get the declared types have pretty varied dependencies, so I didn't think of a good way to come up with a contract for this.